File: app/src/main/kotlin/tech/mmarca/openvitals/domain/query/HeartPeriodData.kt
package tech.mmarca.openvitals.domain.query

import tech.mmarca.openvitals.domain.model.DailyHrv
import tech.mmarca.openvitals.domain.model.DailyRestingHR
import tech.mmarca.openvitals.domain.model.HeartRateSample
import tech.mmarca.openvitals.domain.model.HeartRateSummary

data class HeartPeriodData(
    val daySamples: List<HeartRateSample> = emptyList(),
    val previousDaySamples: List<HeartRateSample> = emptyList(),
    val dailySummaries: List<HeartRateSummary> = emptyList(),
    val previousDailySummaries: List<HeartRateSummary> = emptyList(),
    val baselineDailySummaries: List<HeartRateSummary> = emptyList(),
    val dayRestingBpm: Long? = null,
    val previousDayRestingBpm: Long? = null,
    val dayHrvMs: Double? = null,
    val previousDayHrvMs: Double? = null,
    val dailyRestingHR: List<DailyRestingHR> = emptyList(),
    val previousDailyRestingHR: List<DailyRestingHR> = emptyList(),
    val baselineDailyRestingHR: List<DailyRestingHR> = emptyList(),
    val dailyHrv: List<DailyHrv> = emptyList(),
    val previousDailyHrv: List<DailyHrv> = emptyList(),
    val baselineDailyHrv: List<DailyHrv> = emptyList(),
)

File: lib/domain/query/heart_period_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../model/heart_models.dart';

part 'heart_period_data.freezed.dart';

@freezed
abstract class HeartPeriodData with _$HeartPeriodData {
  const factory HeartPeriodData({
    @Default(<HeartRateSample>[]) List<HeartRateSample> daySamples,
    @Default(<HeartRateSample>[]) List<HeartRateSample> previousDaySamples,
    @Default(<HeartRateSummary>[]) List<HeartRateSummary> dailySummaries,
    @Default(<HeartRateSummary>[]) List<HeartRateSummary> previousDailySummaries,
    @Default(<HeartRateSummary>[]) List<HeartRateSummary> baselineDailySummaries,
    @Default(<RestingHeartRateSample>[])
    List<RestingHeartRateSample> dayRestingSamples,
    int? dayRestingBpm,
    int? previousDayRestingBpm,
    @Default(<HrvSample>[]) List<HrvSample> dayHrvSamples,
    double? dayHrvMs,
    double? previousDayHrvMs,
    @Default(<DailyRestingHR>[]) List<DailyRestingHR> dailyRestingHR,
    @Default(<DailyRestingHR>[]) List<DailyRestingHR> previousDailyRestingHR,
    @Default(<DailyRestingHR>[]) List<DailyRestingHR> baselineDailyRestingHR,
    @Default(<DailyHrv>[]) List<DailyHrv> dailyHrv,
    @Default(<DailyHrv>[]) List<DailyHrv> previousDailyHrv,
    @Default(<DailyHrv>[]) List<DailyHrv> baselineDailyHrv,
  }) = _HeartPeriodData;
}
